pkg/asset/agent/manifests/nmstateconfig.go

```diff
@@ -57,14 +60,53 @@ func (*NMStateConfig) Name() string {
 // the asset.
 func (*NMStateConfig) Dependencies() []asset.Asset {
 	return []asset.Asset{
-		// TODO - use the asset to provide input for nmstateconfig
+		&agentconfig.AgentConfig{},
 	}
 }
 
 // Generate generates the NMStateConfig manifest.
 func (n *NMStateConfig) Generate(dependencies asset.Parents) error {
 
-	// TODO - will generate nmstateconfig from the asset used to provide input
+	agentConfig := &agentconfig.AgentConfig{}
+	dependencies.Get(agentConfig)
+
+	nmStateConfigs := []*aiv1beta1.NMStateConfig{}
+
+	if agentConfig.Config != nil {
+		for _, host := range agentConfig.Config.Spec.Hosts {
+			nmStateConfig := aiv1beta1.NMStateConfig{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "NMStateConfig",
+					APIVersion: "agent-install.openshift.io/v1beta1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      agentConfig.Config.ObjectMeta.Name,
+					Namespace: agentConfig.Config.ObjectMeta.Namespace,
+					Labels: map[string]string{
+						"infraenvs.agent-install.openshift.io": "ostest",
+					},
+				},
+				Spec: aiv1beta1.NMStateConfigSpec{
```

Comment:
Need to add `Interfaces` inside the Spec here.

Reply:
Also, I hardcoded the label. Need to fix it.

Reply:
Both are addressed.